examples/sdl-audio.rs

```diff
@@ -0,0 +1,135 @@
+//! # Example: SDL Audio
+//!
+//! This example demonstrates how SDL can be used not only to implement virtual displays, but at the same time
+//! to use it as an audio device. Here we implement an oscillator with a modulation of its pitch.
+
+extern crate embedded_graphics;
+extern crate embedded_graphics_simulator;
+
+use std::sync::{
+    atomic::{AtomicBool, Ordering},
+    Arc,
+};
+
+use embedded_graphics::{
+    mono_font::{ascii::FONT_6X10, MonoTextStyle},
+    pixelcolor::BinaryColor,
+    prelude::*,
+    text::Text,
+};
+use embedded_graphics_simulator::{
+    sdl2::Keycode, OutputSettingsBuilder, SimulatorDisplay, SimulatorEvent, Window,
+};
+use sdl2::audio::{AudioCallback, AudioSpecDesired};
+
+const SAMPLE_RATE: i32 = 44100;
+
+const PITCH_MIN: f32 = 440.0;
+const PITCH_MAX: f32 = 10000.0;
+
+const PERIOD: f32 = 0.5; // seconds
+const SAMPLES_PER_PERIOD: f32 = SAMPLE_RATE as f32 * PERIOD;
+const PITCH_CHANGE_PER_SAMPLE: f32 = (PITCH_MAX - PITCH_MIN) / SAMPLES_PER_PERIOD;
+
+fn main() -> Result<(), core::convert::Infallible> {
+    // Prepare the audio "engine" with gate control
+    let gate = Arc::new(AtomicBool::new(false));
+    let audio_wrapper = AudioWrapper::new(gate.clone());
+
+    let audio_spec = AudioSpecDesired {
+        freq: Some(SAMPLE_RATE),
+        channels: Some(1),
+        samples: Some(32),
+    };
+
+    // This is the initialisation of SDL and capturing its audio device.
+    let audio_device = sdl2::init()
+        .unwrap()
+        .audio()
+        .unwrap()
+        .open_playback(None, &audio_spec, |_| audio_wrapper)
+        .unwrap();
+
+    audio_device.resume();
```

Comment:
I think we could elaborate this a bit more. What do you think about this?
```suggestion
    // Initialize the SDL audio subsystem.
    //
    // `sdl2` allows multiple instances of the SDL context to exist, which makes
    // it possible to access SDL subsystems which aren't used by the simulator.
    // But keep in mind that only one `EventPump` can exists and the simulator
    // window creation will fail if the `EventPump` is claimed in advance.
    let sdl = sdl2::init().unwrap();
    let audio_subsystem = sdl.audio().unwrap();

    // Start audio playback by opening the device and setting the custom callback.
    let audio_device = audio_subsystem.open_playback(None, &audio_spec, |_| audio_wrapper).unwrap();
    audio_device.resume();
```

Reply:
sounds good!